chapter-outline.md

```diff
@@ -0,0 +1,287 @@
+              (although the more I think about it the more it enthralls me =p).
+- As part of the grant proposal I wrote about creating videos to enhance students learning.
+    - I don't think we are strictly bound by this,
+      but if we can think of ways where this complements what students can learn from the text
+      then I would like to include it
+      as I think it can also mix up the content more
+      and engage via different modalities.
+        - My initial idea for these videos is around me drawing charts and concepts.
+          I think the process of drawing
+          is something that students cannot get from a static resource,
+          since that will just show the output.
+          It could shed light on some conceptual aspect
+          and allows students to see aspect laid out more visually.
+          For example,
+          I drew out a schematic for how all git commands are connecting
+          when I originally taught 521,
+          and it was really helpful for students to get a graphical overview of what is happening.
+- Should we enable comments on the entire book? There are technologies such as `hypothes.is`
+  that are easily enabled in quarto
+  and allows student to highlight sections in the book and take notes.
+  They can even opt to make their notes public to other students
+  and have discussions in the comments
+  (maybe preferential and more direct vs having discussion on slack?),
+  but there would be no notifications and we are splitting discussion into two avenues.
+  Maybe more a space to exchange reflections on the content?
+- Should there be a dedicated section for common pitfalls?
+  This seems hard to execute and I would prefer to bring up pitfalls
+  as we go through each section in the chapter
+  rather than collect them all in a single section.
+    - An alternative to simply bringing it up in the text would be to
+      have callout boxes so it is easy to identify the pitfalls for each chapters,
+      but they are also occurring right after the section they are related to.
+        - Maybe this organization into different callout boxes (or similar semantic unit)
+          would allow us to have alternative ways of viewing the content.
+          E.g., maybe there could be a way to collate all callout boxes into a single page
+          (and like-wise all learning objectives, all excericese, etc separate pages)
+          either "on demand" or during the build process of the book.
+- Should we have some type of "review what you have learned so far" elements/chapters
+  every nth chapter or is that too much hand-holding
+  and too much work for us in addition to the existing exercises in each chapter?
+  Can it be done in a fun way?
+- Should we have a "how to use this book" chapter?
+  Or even lay out the philosophy behind some of our choices?
+  Maybe at least a short version in some type of foreword,
+  but I don't want to students to have to read a manual-like compilation
```

Comment:
Makes sense if we are transcending a typical textbook. Given that most of the elements we have discussed are visual most of this how to use could visual too like an annotated summary of the motifs we use in the book.

Reply:
The suggestion here is to create a diagram/visual guide of the motifs/callout boxes in the book. An "IKEA" manual rather than a text heavy one.

---

Ideally, the elements of the chapters would be self-explanatory and we don't need this section.